model/artifact/artifact_file.go

```diff
@@ -113,10 +115,35 @@ func presignFile(ctx context.Context, file File) (string, error) {
 
 	// If this bucket is a devprod owned one, we sign the URL
 	// with the app's server IRSA credentials (which is used
-	// when no credentials are provided).
+	// when no credentials are provided). If it fails,
+	// we fallback to using the provided credentials.
 	if isInternalBucket(file.Bucket) {
-		file.AwsKey = ""
-		file.AwsSecret = ""
+		requestParams := pail.PreSignRequestParams{
+			Bucket:                file.Bucket,
+			FileKey:               file.FileKey,
+			SignatureExpiryWindow: evergreen.PresignMinimumValidTime,
+		}
+		presignURL, err := pail.PreSign(ctx, requestParams)
+		if err != nil {
+			return "", errors.Wrap(err, "presigning internal bucket file")
+		}
+		// Test if the presigned URL is valid by making a request to it.
+		resp, err := http.Get(presignURL)
```

Comment:
If possible, long-lived production processes like the agent or app server should prefer to use a `utility.GetHTTPClient` instead of the `http.DefaultClient` like `http.Get` does. Historically, we've attempted to get rid of all usages of the default client because the default client has undesirable behavior for a production application.

You can read more about it [here](https://dev.to/gkampitakis/http-connection-churn-in-go-34pl) (there's more reasons, but the main ones are mentioned there).

Reply:
The usage looks right to me!